src/java/org/apache/cassandra/index/sai/disk/v1/postings/PostingListRangeIterator.java

```diff
@@ -83,10 +82,23 @@ public PostingListRangeIterator(IndexContext indexContext,
         this.indexContext = indexContext;
         this.primaryKeyMap = primaryKeyMap;
         this.postingList = searcherContext.postingList;
-        this.searcherContext = searcherContext;
+        this.rowIdOffset = searcherContext.segmentRowIdOffset;
         this.queryContext = searcherContext.context;
     }
 
+    public PostingListRangeIterator(IndexContext indexContext,
+                                    PrimaryKeyMap primaryKeyMap,
+                                    PostingList postingList,
+                                    QueryContext queryContext)
+    {
+        super(primaryKeyMap.primaryKeyFromRowId(postingList.minimum()), primaryKeyMap.primaryKeyFromRowId(postingList.maximum()), postingList.size());
```

Comment:
Nit: break the long line
```suggestion
        super(primaryKeyMap.primaryKeyFromRowId(postingList.minimum()), 
              primaryKeyMap.primaryKeyFromRowId(postingList.maximum()), 
              postingList.size());
```